src/Output.cpp

```diff
@@ -0,0 +1,109 @@
+#include "CodeGen_C.h"
+#include "StmtToHtml.h"
+#include "Output.h"
+#include "LLVM_Headers.h"
+#include "LLVM_Output.h"
+
+#include <fstream>
+
+namespace Halide {
+
+void output_object(const Module &module, std::string filename) {
+    if (filename.empty()) filename = module.name() + ".o";
+
+    llvm::LLVMContext context;
+    llvm::Module *llvm = output_llvm_module(module, context);
+    output_object(llvm, filename);
+    delete llvm;
+}
+
+void output_assembly(const Module &module, std::string filename)  {
+    if (filename.empty()) filename = module.name() + ".s";
+
+    llvm::LLVMContext context;
+    llvm::Module *llvm = output_llvm_module(module, context);
+    output_assembly(llvm, filename);
+    delete llvm;
+}
+
+void output_native(const Module &module,
+                   std::string object_filename,
+                   std::string assembly_filename) {
+    if (object_filename.empty()) object_filename = module.name() + ".o";
+    if (assembly_filename.empty()) assembly_filename = module.name() + ".s";
+
+    llvm::LLVMContext context;
+    llvm::Module *llvm = output_llvm_module(module, context);
+    output_object(llvm, object_filename);
```

Comment:
I found it confusing that there's output_object(llvm::Module *, string) and also output_object(Halide::Module, string).